docs/source/examples/performance-training.rst

```diff
@@ -0,0 +1,408 @@
+
+Example configuration:
+
+.. code-block:: yaml
+
+    file_mounts:
+      /checkpoint:
+        name: my-checkpoint-bucket  
+        mode: MOUNT_CACHED
+
+.. note::
+   
+   In MOUNT_CACHED mode, files only begin uploading after they are closed by all processes. By default, SkyPilot uses sequential transfers to maintain file order, which can cause the cache to grow if files are written faster than they can be uploaded.
+
+   When multiple jobs share a node (e.g., using fractional GPUs), be aware of these important considerations:
+   
+   - A job is only marked complete after all files in the cache are flushed to the remote bucket, including those from other jobs.
+   - The total disk space required equals the sum of all jobs' cached data.
+   - Jobs that finish quickly might need to wait for other jobs' files to be uploaded.
+   - Ensure the disk tier is fast enough to handle the combined write load of all jobs.
+
+To see the difference between MOUNT and MOUNT_CACHED, see :ref:`storage mounting modes <storage-mounting-modes>`.
+
+Robust checkpointing for spot instances
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using spot instances, robust checkpointing is crucial for recovering from preemptions. 
+
+To make checkpointing reading robust against preemptions, use this recipe:
+
+- Always try loading from the latest checkpoint first 
+- If the latest checkpoint is found to be corrupted or incomplete,  fallback to earlier checkpoints
+
+Here's a simplified example showing the core concepts:
+
+.. code-block:: python
+
+    def load_checkpoint(save_dir: str):
+        try:
+            # Find all checkpoints, sorted by step (newest first)
+            checkpoints = sorted(
+                [f for f in Path(save_dir).glob("checkpoint_*.pt")],
+                key=lambda x: int(x.stem.split('_')[-1]),
+                reverse=True
+            )
+            
+            # Try each checkpoint from newest to oldest
+            for checkpoint in checkpoints:
+                try:
+                    step = int(checkpoint.stem.split('_')[-1])
+                    result = load_checkpoint(checkpoint) # need to fill in
+                    return result
+                except Exception as e:
+                    logger.warning(f"Failed to load checkpoint {step}: {e}")
+                    continue
+        except Exception as e:
+            logger.error(f"Failed to find checkpoints: {e}")
+            return None
+
+For a complete implementation with additional features like custom prefixes, extended metadata, and more detailed error handling, see the code below:
+
+.. dropdown:: Full Implementation
+    :animate: fade-in-slide-down
+
+    .. code-block:: python
```

Comment:
TBH, I think this is a lot more complicated than it needs to be.
1. The only thing that the `@save_checkpoint` does is write the metadata files, but we don't read them. Is it really necessary?
2. `@load_checkpoint` decorates a function that takes a `step`, but you don't actually need to pass in the `step` when creating it. I'm not sure it makes a lot of sense as a decorator.

To be honest, I think it may be better just to stick to the short snippet above. I'm worried this longer snippet may intimidate people and doesn't provide a lot of value.

Reply:
This is resolved by https://github.com/skypilot-org/skypilot/pull/5100#discussion_r2025555199